pkg/promtail/client/batch.go

```diff
@@ -0,0 +1,86 @@
+package client
+
+import (
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/prometheus/common/model"
+)
+
+type batch struct {
```

Comment:
Comment would be useful here. On first read (before seeing client.go), I didn’t understand why is this needed. (After seeing client.go I can see that this is a generalization of code that was previously there)

Reply:
Nice explanation! Thanks.